starfish/codebook.py

```diff
@@ -356,13 +358,65 @@ def to_json(self, filename: str) -> None:
         with open(filename, 'w') as f:
             json.dump(code_array, f)
 
-    def decode_euclidean(self, intensities: IntensityTable) -> IntensityTable:
+    @staticmethod
+    def _unit_normalize(array: xr.DataArray) -> xr.DataArray:
+        """unit normalize each feature of array"""
+        total_intensities = array.sum(dim=[Indices.CH.value, Indices.HYB.value])
+        array = array / total_intensities
+
+        # if a code is all zero, the unit intensity should be spread across the channel
```

Comment:
this comment is weird because this is used for both normalizing the intensity table and the codebook.  does the same policy apply to both?  guessing yes, given the nature of the code.